services/comparison/overlays/critical_path.py

```diff
@@ -1,20 +1,26 @@
 import json
 import re
-from typing import Sequence
+from typing import Self, Sequence
 
 from asgiref.sync import async_to_sync
 from cc_rustyribs import rustify_diff
 from shared.profiling import ProfilingDataFullAnalyzer, ProfilingSummaryDataAnalyzer
 from shared.storage.exceptions import FileNotInStorageError
 
+from database.models.core import GITHUB_APP_INSTALLATION_DEFAULT_NAME
 from database.models.profiling import ProfilingCommit
 from services.repository import get_repo_provider_service
 from services.yaml import get_current_yaml
 
 sentinel = object()
 
 
-def _get_latest_profiling_commit(comparison):
+def _get_latest_profiling_commit(comparison) -> ProfilingCommit | None:
+    """
+    @param comparison: ComparisonProxy (not imported due to circular imports)
```

Comment:
You can move that import behind a  `if typing.TYPE_CHECKING:` guard.
See https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING and https://stackoverflow.com/questions/61545580/how-does-mypy-use-typing-type-checking-to-resolve-the-circular-import-annotation

Same below as well.